main.go

```diff
@@ -2,16 +2,30 @@ package main
 
 import (
 	"./logging"
+	"fmt"
 	"log"
 	"os"
 	"strings"
+	"sync"
+
+	"github.com/hashicorp/go-reap"
 )
 
 func main() {
 	log.SetOutput(os.Stdout)
 	if strings.EqualFold(os.Getenv("DEBUG"), "true") {
 		go logging.StartLogging()
 	}
+	// Get feedback on reaped children and errors.
+	if reap.IsSupported() {
```

Comment:
I don't think there's a need for the conditional. Since we're running everything in a container (controlled environment) and the only processes that should be reaped belong to HAProxy, we can assume that it is always the same condition. In our scenario, reaping is either always or never supported. If it is not supported (haven't tried it myself) the whole idea of using this library would fail. Assuming that reaping HAProcesses is supported, I'd suggest removing the conditionals.